packages-exp/auth-exp/src/core/persistence/browser.ts

```diff
@@ -56,9 +56,28 @@ class BrowserPersistence implements Persistence {
   }
 }
 
-export const browserLocalPersistence: externs.Persistence = new BrowserPersistence(
-  localStorage
-);
-export const browserSessionPersistence: externs.Persistence = new BrowserPersistence(
-  sessionStorage
-);
+class BrowserLocalPersistence {
+  static type: 'LOCAL' = 'LOCAL';
+  static instance: Persistence | null = null;
+  static _getInstance(): Persistence {
+    if (!this.instance) {
+      this.instance = new BrowserPersistence(localStorage);
+    }
+    return this.instance;
+  }
+}
+
+class BrowserSessionPersistence {
```

Comment:
shouldn't these implement `PersistenceInstantiator` to make the cast safe?

Reply:
Obsolete; there's no way to tell TS "hey, this class's static instance implements these methods" unfortunately